src/video/wayland/SDL_waylandwindow.c

```diff
@@ -977,6 +1129,13 @@ Wayland_SetWindowMaximumSize(_THIS, SDL_Window * window)
     SDL_WindowData *wind = window->driverdata;
 
     if (window->flags & SDL_WINDOW_RESIZABLE) {
+#ifdef HAVE_LIBDECOR_H
+        if (data->shell.libdecor) {
+            libdecor_frame_set_max_content_size(wind->shell_surface.libdecor.frame,
+                                                window->max_w,
+                                                window->max_h);
+        }
+#else
```

Comment:
```suggestion
        } else
#endif
```
Along with removing the `#endif` slightly below.